### PR TITLE
Site visibility settings add documentation link and copy changes

### LINKED ISSUE #50779: Add documentation link to site visibility settings
**Prerequisites (mark completed items with an [x]):**
- [ ] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [ ] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**

To make the documentation easier to find, let's try adding it to the intro text of site visibility settings.

image-38

## Acceptance criteria

* Add "Learn more" link to site visibility settings intro text.
* Link to `https://woocommerce.com/document/configuring-woocommerce-settings/coming-soon-mode/` in a new tab.

### PR DESCRIPTION
**Pull request by ilyasfoo**
### Changes proposed in this Pull Request:

Closes #50779

Minor copy changes and add documentation link.

image

### How to test the changes in this Pull Request:

1. Go to `WooCommerce > Settings > Site visibility`
2. Observe the copy changes highlighted in screenshot:
    - `Manage how your site appears to visitors. Learn more`
    - `Apply to store pages only`
    - Removed link from `store pages`
3. Click on `Learn more`
4. Observe `https://woocommerce.com/document/configuring-woocommerce-settings/coming-soon-mode/` is opened in a new tab
5. Click on `Editor`
6. Observe it opens up site editor in a new tab

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/launch-your-store/settings/slotfill.js [hunk 1]
```diff
@@ -183,20 +193,13 @@ const SiteVisibility = () => {
...
-											'Display a "coming soon" message on your <a>store pages</a> — the rest of your site will remain visible.',
-											'woocommerce'
-										),
-										{
-											a: createElement( 'a', {
-												href: SITE_VISIBILITY_DOC_LINK,
-											} ),
-										}
+									{ __(
+										'Display a “coming soon” message on your store pages — the rest of your site will remain visible.',
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/launch-your-store/settings/slotfill.js:201]: Should we use `"` or `“` and `”` for the quotes? I see that the original string uses `"`?  It seems that we're using both of them in the codebase. 😅

**Review by chihsuan — APPROVED** [review]
Thanks, tested well! Just left a minor comment, and I think we need to update e2e tests as well. 🙂 

Pre-approving! ✅

File: plugins/woocommerce-admin/client/launch-your-store/settings/slotfill.js (hunk 1 above)
Comment (ilyasfoo) [reply plugins/woocommerce-admin/client/launch-your-store/settings/slotfill.js:201]: @chihsuan Yeah I assumed the design was inaccurate (clarified it in Figma as well). Some discussion regarding updating it here p1724125186383239/1724116284.730329-slack-C01SFMVEYAK

**Review by adrianduffell — APPROVED** [review]
LGTM! Thanks @ilyasfoo 🚀

**Pull request merged**